src/js/page_sections/I18nBanner.js

```diff
@@ -1,50 +1,34 @@
-import React, { Component } from 'react';
-import locale from 'browser-locale';
+import React, { Component } from 'react'
+import { Link } from 'react-router-dom'
 
+import SUPPORTED_LANGUAGES from 'js/constants/languages'
 import CaretDownSVG from 'js/svg/CaretDown';
 
-const languages = [
-  {
-    title: 'English',
-    abbr: 'en',
-  }, {
-    title: 'Español',
-    abbr: 'es',
-  }, {
-    title: 'Tiếng Việt',
-    abbr: 'vn',
-  }, {
-    title: '中文',
-    abbr: 'zh',
-  }, {
-    title: 'عربى',
-    abbr: 'ar',
-  }, {
-    title: '한국어',
-    abbr: 'ko',
-  }, {
-    title: 'اردو',
-    abbr: 'ur',
-  }, {
-    title: 'မြန်မာ',
-    abbr: 'my',
-  },
-]
 
 class I18nBanner extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      activeLanguage: locale().substring(0,2) || 'en',
+      isOpen: false,
     };
     this.displayedLanguageAmount = 4;
-    this.visibleLanguageOptions = languages.slice(0, this.displayedLanguageAmount);
-    this.secondaryLanguageOptions = languages.slice(this.displayedLanguageAmount, languages.length);
+    this.visibleLanguageOptions = SUPPORTED_LANGUAGES.slice(0, this.displayedLanguageAmount);
+    this.secondaryLanguageOptions = SUPPORTED_LANGUAGES.slice(this.displayedLanguageAmount, SUPPORTED_LANGUAGES.length);
+    console.log(this.getPathname())
+  }
+
+  getPathname = () => {
+    let pathArray = this.props.location.pathname.split('/').filter(n => n)
+    if (SUPPORTED_LANGUAGES.map(lang => lang.code).includes(pathArray[0])){
```

Comment:
How would you feel about pre-calculating `SUPPORTED_LANG_CODES` inside of `constants.js`. It would save the recalculation in other places, like https://github.com/cityofaustin/janis/pull/74/files#diff-05c601d58da6bea7d129c7ef04b9e2feR33

Reply:
i like it!